app/js/MongoManager.js

```diff
@@ -64,21 +64,29 @@ module.exports = MongoManager = {
     db.docs.find(query, queryOptions).toArray(callback)
   },
 
-  getArchivedProjectDocs(project_id, callback) {
+  getNArchivedProjectDocs(project_id, n, callback) {
     const query = {
       project_id: ObjectId(project_id.toString()),
       inS3: true
     }
-    db.docs.find(query).toArray(callback)
+    db.docs.find(query, { projection: { _id: 1 }, limit: n }).toArray(callback)
   },
 
-  getNonDeletedArchivedProjectDocs(project_id, callback) {
+  getNNonArchivedProjectDocs(project_id, n, callback) {
```

Comment:
Stylistically, I think getNNonArchivedProjectDocs is a bit unusual. I think it's OK to stick with the original method name and just require the additional max-results parameter?

Also, I think `n` isn't very clear - what about `maxResults`? e.g.

```JavaScript
  getNonArchivedProjectDocs(project_id, maxResults, callback) {
```

Reply:
Done in https://github.com/overleaf/docstore/pull/109/commits/d1fb8530b95c7394e7d581cc07ee850b2438967b